### PR TITLE
BaseAutofill: move to be it's own component

### PR DESCRIPTION
**Pull request by joschect**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

BaseAutofill: move to be it's own component. Also renamed it to be properly cased.

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -329,8 +329,8 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
             spellCheck={ false }
             defaultVisibleValue={ this._currentVisibleValue }
             suggestedDisplayValue={ suggestedDisplayValue }
-            updateValueInWillReceiveProps={ this._onUpdateValueInAutoFillWillReceiveProps }
-            shouldSelectFullInputValueInComponentDidUpdate={ this._onShouldSelectFullInputValueInAutoFillComponentDidUpdate }
+            updateValueInWillReceiveProps={ this._onUpdateValueInAutofillWillReceiveProps }
+            shouldSelectFullInputValueInComponentDidUpdate={ this._onShouldSelectFullInputValueInAutofillComponentDidUpdate }
```
Comment (lambertwang) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Is there a shorter name that can convey the purpose of this variable?

File: packages/office-ui-fabric-react/src/components/Autofill/BaseAutofill.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+} from '../../Utilities';
+
+export interface IBaseAutofillState {
+  displayValue?: string;
+}
+
+const SELECTION_FORWARD = 'forward';
+const SELECTION_BACKWARD = 'backward';
+
+export class BaseAutofill extends BaseComponent<IBaseAutofillProps, IBaseAutofillState> implements IBaseAutofill {
```
Comment (amyngu) [inline packages/office-ui-fabric-react/src/components/Autofill/BaseAutofill.tsx]: Should these be changed to AutoFill also?

Comment (joschect) [reply packages/office-ui-fabric-react/src/components/Autofill/BaseAutofill.tsx]: Maybe, that's a good idea.

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (joschect) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Maybe, I don't remember when this got added. I don't mind long names, they get compiled down!

**Pull request merged**